pkg/utils/utils.go

```diff
@@ -112,7 +124,16 @@ func DiscoverSriovDevices() ([]sriovnetworkv1.InterfaceExt, error) {
 				}
 			}
 		}
-		pfList = append(pfList, iface)
+
+		for _, devName := range deviceNames {
+			for _, nic := range net.NICs {
+				if devName == nic.Name {
+					glog.V(2).Infof("DiscoverSriovDevices() adding device %s, pci address %s\n", nic.Name, device.Address)
+					pfList = append(pfList, iface)
+					break
+				}
+			}
+		}
```

Comment:
Im trying to wrap my head around the changes in this PR.

looking at https://github.com/jaypipes/ghw/blob/14923eaaa4d8378051f741694e261747d16fc05e/net_linux.go#L43

the `NICs` attribute is essentially a list of all netdevices in the host built from `/sys/class/net`

`deviceNames` is yet another list of netdevices found for a specific pci device by scanning `/sys/bus/pci/devices/<pci addr>/net`

now, in my system, moving a netdev to a different namespace will remove it from both `/sys/class/net` and `/sys/bus/pci/devices/<pciAddr>/net`  (as they are both the same file under `/sys/devices/....`)

So it sifts down to the fact that the latter is a subset of the former, am i missing something ?

essentially the one case where you would not want `pfList` to contain a certain `iface` (device) is if it has no netdevs associated with it. so you could just check `len(deviceNames) == 0` at L92

Reply:
You totally right, testing and updating according what you suggested.

Reply:
Done